pywhy_stats/fisherz.py

```diff
@@ -0,0 +1,129 @@
+"""Independence test using Fisher-Z's test.
+
+This test is also known as the partial correlation independence test.
+It works on Gaussian random variables.
+
+When the data is not Gaussian, this test is not valid. In this case, we recommend
+using the Kernel independence test at <insert link>.
+
+Examples
+--------
+>>> import pywhy_stats as ps
+>>> res = ps.fisherz.ind([1, 2, 3], [4, 5, 6])
+>>> print(res.pvalue)
+>>> 1.0
+"""
+
+from math import log, sqrt
+from typing import Optional
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy.stats import norm
+
+from .pvalue_result import PValueResult
+
+
+def ind(X: ArrayLike, Y: ArrayLike, correlation_matrix: Optional[ArrayLike] = None) -> PValueResult:
+    """Perform an independence test using Fisher-Z's test.
+
+    Works on Gaussian random variables. This test is also known as the
+    correlation test.
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples,)
+        The first node variable.
+    Y : ArrayLike of shape (n_samples,)
+        The second node variable.
+    correlation_matrix : ArrayLike of shape (2, 2), optional
+        The precomputed correlation matrix between X and Y., by default None.
+
+    Returns
+    -------
+    X : float
+        The test statistic.
+    p : float
+        The p-value of the test.
+    """
+    return _fisherz(X, Y, condition_on=None, correlation_matrix=correlation_matrix)
+
+
+def condind(
+    X: ArrayLike,
+    Y: ArrayLike,
+    condition_on: ArrayLike,
+    correlation_matrix: Optional[ArrayLike] = None,
+) -> PValueResult:
+    """Perform an independence test using Fisher-Z's test.
+
+    Works on Gaussian random variables. This test is also known as the
+    correlation test.
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples,)
+        The first node variable.
+    Y : ArrayLike of shape (n_samples,)
+        The second node variable.
+    condition_on : ArrayLike of shape (n_samples, n_variables)
+        The conditioning set.
+    correlation_matrix : ArrayLike of shape (2 + n_variables, 2 + n_variables), optional
+        The precomputed correlation matrix between X, Y and ``condition_on``, by default None.
+
+    Returns
+    -------
+    X : float
+        The test statistic.
+    p : float
+        The p-value of the test.
+    """
+    return _fisherz(X, Y, condition_on=condition_on, correlation_matrix=correlation_matrix)
+
+
+def _fisherz(
+    X: ArrayLike,
+    Y: ArrayLike,
+    condition_on: Optional[ArrayLike] = None,
+    correlation_matrix: Optional[ArrayLike] = None,
+) -> PValueResult:
+    """Perform an independence test using Fisher-Z's test.
```

Comment:
I would say that here the docstring for the `_` methods is rather optional. It's fine to have it, but we don't really need it (especially, since this would be another copy paste of the same docstring :))

Reply:
I added it more for the developer's sake. I usually follow the in-file docstrings to understand code. It should be a fairly simple ask to copy/paste the master docstring, unless something changes with the function arguments in each `_` methods. If it does change, then it would be nice to document that in the docstring anyways.